src/backlight/portfolio/portfolio.py

```diff
@@ -72,24 +72,70 @@ def construct_portfolio(
         Portfolio
     """
 
+    assert len(principal) == len(trades)
+    assert len(lot_size) == len(trades)
+
     symbols2mkt = {m.symbol: m for m in mkt}
     symbols = [t.symbol for t in trades]
     assert set(symbols) == set(symbols2mkt.keys())
 
     # Transform trades following the lot_size
-    for (trade, lot) in zip(trades, lot_size):
-        trade["amount"] *= lot
```

Comment:
What is the reason to replace implementation here?

Reply:
I just remember the reason. Doing it this way was modifying the values inside the `trades` array even outside the function, which is usually something to avoid. Maybe I'm wrong here.

Reply:
I see. That is true.